build-logic/conventions/src/main/kotlin/com/squareup/anvil/conventions/PublishConventionPlugin.kt

```diff
@@ -131,7 +148,7 @@ open class PublishExtension @Inject constructor(
 
         // Gradle plugin publications have their own artifactID convention,
         // and that's handled automatically.
-        if (!publication.name.endsWith("PluginMarkerMaven")) {
+        if (!publication.name.endsWith("PluginMarkerMaven") && overrideArtifactId) {
```

Comment:
You can probably get rid of the `overrideArtifactId` and replace it with `project.pluginManager.hasPlugin("java-gradle-plugin")`.